api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,1232 @@
+      reception, but they in general MIGHT be set by the client.
+    properties:
+      metadata:
+        type: object
+        properties:
+          eid:
+            description: |
+              Identifier of this Event.
+
+              Clients are allowed to generate this and this SHOULD be guaranteed to be unique from the
+              perspective of the producer. Consumers MIGHT use this value to assert uniqueness of reception
+              of the Event.
+
+              If not present upon reception, Nakadi will generate and fill this with a unique value.
+              (TBD: enforce UUID formatif set by the client?)
+            type: string
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          event_type:
+            description: |
+              The EventType of this Event. This is enriched by Nakadi on reception of the Event based on the
+              endpoint where the Producer sent the Event to.
+
+              If provided MUST match the endpoint. Failure to do so will cause rejection of the Event.
+            type: string
+            example: 'pennybags:payment-business-event'
+          occurred_at:
+            description: |
+              Timestamp of creation of the Event generated by the producer. Presence is optional but strongly
+              recommended for producers.
+            type: string
+            format: date-time
+          received_at:
+            type: string
+            description: |
+              Timestamp of the reception of the Event by Nakadi. This is enriched upon reception of the
+              Event.
+              If set by the producer Event will be rejected.
+            format: date-time
+          parent_eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the Event that caused the generation of this Event.
+              Set by the procuder.
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          root_eid:
+            type: string
+            format: uuid
+            description: |
+              Event identifier of the root of the tree of Events (from the relation constructed from
+              parent-eid)
+            example: '105a76d8-db49-4144-ace7-e683e8f4ba46'
+          flow_id:
+            description: |
+              The flow-id of the producer of this Event. As this is usually a HTTP header, this is enriched
+              from the header into the metadata by Nakadi to avoid clients having to explicitly copy this.
+            type: string
+            example: 'JAh6xH4OQhCJ9PutIV_RYw'
+
+        additionalProperties: false
+
+    required:
+      - metadata
+
+  BusinessEvent:
+    type: object
+    description: |
+      A Business Event.
+
+      Usually represents a status transition in a Business process.
+    allOf:
+      - $ref: '#/definitions/EventMetadata'
+    additionalProperties: true
+
+  DataChangeEventQualifier:
```

Comment:
This turns internal data schemes into APIs – making it even harder to incrementally build software. I especially want to point out that for example the stock data will not evolve in a backwards compatible way, most probably even completely switching technologies for storage. But on the other hand we will of course try to be as backward as possible on API level.

I strongly urge to accept our RESTful resources as "the only truth" and base all other means of data access on these.
If any data is needed for analytical use cases it might also be needed for operational use cases at some point – so let's just have it in our resources.